### PR TITLE
Updated doc on how to enable Symfony Reverse Proxy

### PR DESCRIPTION
**Pull request by vidarl**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | N/A
| Versions      | 3.3, 4.5
| Edition       | Content/Headless, Experience, Commerce

@barw4 detected that the doc for how to enable symfony reverse proxy was outdated.

Here is a PR that corrects the issue.

#### Checklist

- [ ] Text renders correctly
- [ ] Text has been checked with vale
- [ ] Description metadata is up to date
- [ ] Redirects cover removed/moved pages
- [ ] Code samples are working
- [ ] PHP code samples have been fixed with PHP CS fixer
- [ ] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

**Comment by vidarl** [comment]
The diff for 4.5 is different. You'll take care of that while merging, or should I put in a separate PR:

```diff
diff --git a/public/index.php b/public/index.php
index 9982c21..8d7187a 100644
--- a/public/index.php
+++ b/public/index.php
@@ -1,9 +1,11 @@
 <?php

 use App\Kernel;
+use Ibexa\Bundle\HttpCache\AppCache;

 require_once dirname(__DIR__).'/vendor/autoload_runtime.php';

 return function (array $context) {
-    return new Kernel($context['APP_ENV'], (bool) $context['APP_DEBUG']);
+    return new AppCache(new Kernel($context['APP_ENV'], (bool) $context['APP_DEBUG']));
 };
```

**Comment by barw4** [comment]
@vidarl I might be missing something but I'm pretty sure that `public/index.php` by default looks a bit different:
```
<?php

use App\Kernel;

require_once dirname(__DIR__).'/vendor/autoload_runtime.php';

return function (array $context) {
    return new Kernel($context['APP_ENV'], (bool) $context['APP_DEBUG']);
};
```

**Comment by vidarl** [comment]
@barw4 : You are right.
I looked at public/index.php on  my 3.3.33 installation ( which has been updated from 3.3.0 ). Here the index.php looks like it did initially in this PR.
I assume then that my index.php hasn't been updated during the upgrade process for some time.
Installing a fresh 3.3.35 gives me a different index.php. I have updaded this PR accordingly

@dabrt  : Please note that the diff is still different vs 4.5 ( the name space of {{AppCache}} )

**Review by barw4 — APPROVED** [review]

**Review by dabrt — APPROVED** [review]

File: docs/guide/cache/symfony_reverse_proxy.md [hunk 1]
```diff
@@ -4,9 +4,31 @@ description: You can use Symfony HttpCache Proxy, Varnish or Fastly as reverse p
...
+
+ return function (array $context) {
+-    return new Kernel($context['APP_ENV'], (bool) $context['APP_DEBUG']);
++    return new AppCache(new Kernel($context['APP_ENV'], (bool) $context['APP_DEBUG']));
+ };
+```
+
+!!! caution
+
+    Do not enable the Symfony reverse proxy in `public/index.php` if you intend to use Varnish or Fastly.
```
Comment (juskora) [inline docs/guide/cache/symfony_reverse_proxy.md:30]:
```suggestion
    Don't enable the Symfony reverse proxy in `public/index.php` if you intend to use Varnish or Fastly.
```

**Review by juskora — APPROVED** [review]

**Review by julitafalcondusza — APPROVED** [review]

**Pull request merged**